test/common/test_io.cpp

```diff
@@ -197,57 +190,14 @@ TEST (PCL, concatenatePointCloud)
     EXPECT_EQ (cloud_all[cloud_xyz_rgba.size () + i].rgba, cloud_xyz_rgb[i].rgba);
   }
 
-  // _ vs regular
-  int rgb_idx = pcl::getFieldIndex (cloud1, "rgba");
-  cloud1.fields[rgb_idx].name = "_";
-  pcl::concatenatePointCloud (cloud1, cloud2, cloud_out3);
-  
-  pcl::fromPCLPointCloud2 (cloud_out3, cloud_all);
-
-  EXPECT_EQ (cloud_all.size (), cloud_xyz_rgba.size () + cloud_xyz_rgba2.size ());
-  for (int i = 0; i < int (cloud_xyz_rgba.size ()); ++i)
-  {
-    EXPECT_XYZ_EQ (cloud_all[i], cloud_xyz_rgba[i]);
-    // Data doesn't get modified
-    EXPECT_RGB_EQ (cloud_all[i], cloud_xyz_rgba[i]);
-    EXPECT_EQ (cloud_all[i].rgba, cloud_xyz_rgba[i].rgba);
-  }
-  for (int i = 0; i < int (cloud_xyz_rgb.size ()); ++i)
-    EXPECT_XYZ_EQ (cloud_all[cloud_xyz_rgba.size () + i], cloud_xyz_rgb[i]);
-
-  cloud1.fields[rgb_idx].name = "rgba";
-  // regular vs _
-  rgb_idx = pcl::getFieldIndex (cloud2, "rgb");
-  cloud2.fields[rgb_idx].name = "_";
-  pcl::concatenatePointCloud (cloud1, cloud2, cloud_out4);
-
-  pcl::fromPCLPointCloud2 (cloud_out4, cloud_all);
-
-  EXPECT_EQ (cloud_all.size (), cloud_xyz_rgba.size () + cloud_xyz_rgba2.size ());
-  for (int i = 0; i < int (cloud_xyz_rgba.size ()); ++i)
-  {
-    EXPECT_XYZ_EQ (cloud_all[i], cloud_xyz_rgba[i]);
-    // Data doesn't get modified
-    EXPECT_RGB_EQ (cloud_all[i], cloud_xyz_rgba[i]);
-    EXPECT_EQ (cloud_all[i].rgba, cloud_xyz_rgba[i].rgba);
-  }
-  for (int i = 0; i < int (cloud_xyz_rgb.size ()); ++i)
-  {
-    EXPECT_XYZ_EQ (cloud_all[cloud_xyz_rgba.size () + i], cloud_xyz_rgb[i]);
-    EXPECT_FLOAT_EQ (cloud_all[cloud_xyz_rgba.size () + i].r, 0);
-    EXPECT_FLOAT_EQ (cloud_all[cloud_xyz_rgba.size () + i].g, 0);
-    EXPECT_FLOAT_EQ (cloud_all[cloud_xyz_rgba.size () + i].b, 0);
-    EXPECT_EQ (cloud_all[cloud_xyz_rgba.size () + i].rgba, 0);
-  }
-
   // _ vs _
```

Comment:
I'm in favor of a test where we concatenate 2 point clouds: 1 over PointXYZ and one over XYZRGBA with RGBA skipped. If my proposed test is present, remove this test else modify

Reply:
Well, the method doesn't support skipping a field?

It just compares all fields are equal and in the same order, then it copies. Else it reject it if there are any differences, or a "_" field is encountered.

Reply:
It does. It skips "_" in lines 270 and 287

Reply:
The concatenate it self doesn't skip fields when it hit the first "_" in either pointcloud it skips checking the last field of the second pointcloud and hence, it find a missmatch in number of fields and just print this `PCL_ERROR ("[pcl::PCLPointCloud2::concatenate] Number of fields to copy in cloud1 (%u) != Number of fields to copy in cloud2 (%u)\n", i, j); ` and returns false:

https://github.com/PointCloudLibrary/pcl/blob/146ad42cdabaff2b207bd9a3399b8ba0d0b29e7c/common/src/PCLPointCloud2.cpp#L105-L136

Reply:
And if one have a PoinXYZI pointcloud and PointXYZRGBA pointcloud and set both last field names to "_" it would end up doing insert with one kind of data, into a cloud with other data, since it would conclude it is `simplelayout`:
https://github.com/PointCloudLibrary/pcl/blob/146ad42cdabaff2b207bd9a3399b8ba0d0b29e7c/common/src/PCLPointCloud2.cpp#L83-L88

And this skips the entire checking of "_" values, continuing with just copying the data:

https://github.com/PointCloudLibrary/pcl/blob/146ad42cdabaff2b207bd9a3399b8ba0d0b29e7c/common/src/PCLPointCloud2.cpp#L145-L149

Reply:
Can we move this discussion to a separate PR and revert all changes with respect to it there. The issue doesn't seem superficial.